plugins/woocommerce-admin/client/marketplace/components/products/products.tsx

```diff
@@ -127,6 +129,8 @@ export default function Products( props: ProductsProps ): JSX.Element {
 					products={ products }
 					type={ props.type }
 					className={ productListClass }
+					searchTerm={ props.searchTerm }
+					category={ category }
```

Comment:
What if we don't pass this data to components and just add it to every tracking event directly from query?

We could have `recordEvent` in functions to handle the logic.

Reply:
Thanks for the suggestion! I believe I can do this easily... I'll try now.

Reply:
This one wasn't easy. We need `useQuery` for `searchTerm` and `category` but we can't call that hook outside of a React component. So prop drilling looks ugly but it's probably OK for now. It would be good if this search term and category info was in a context or a store.

Reply:
Couldn't fetch `category` and `searchTerm` but created a generic `recordTracksEvent` in product card: https://github.com/woocommerce/woocommerce/pull/41365/commits/b75b8e8a0f199e883a5791623347af54e7437ab1.

Reply:
Nice!